src/tools/rustfmt/src/items.rs

```diff
@@ -1947,6 +1947,10 @@ pub(crate) fn rewrite_struct_field(
     if contains_skip(&field.attrs) {
         return Ok(context.snippet(field.span()).to_owned());
     }
+    // FIXME(default_field_values): Implement formatting.
+    if field.default.is_some() {
+        return Ok(context.snippet(field.span()).to_owned());
```

Comment:
I think we need to change this to `return Err(RewriteError::Unknown);` for now. I don't think the `field.span()` contains the default values. At least I still saw the error reported in https://github.com/rust-lang/rustfmt/issues/6424 when I built rustfmt from this branch and ran it on

```rust
frame_support::construct_runtime!(
    pub struct Test {
        System: frame_system = 0,
        SelfDomainId: pallet_domain_id = 1,
    }
);
```

If possible, could you also include the above snippet in the test case.

Reply:
You're right, the field span doesn't include the optional const. I just didn't actually re-test my test.

I'm not going to include the snippet because it's identical in behavior to the snippet I included, though. I'll adjust the behavior though, and make sure to re-test it.